### ISSUE TITLE
Inference related Queries

### ISSUE THREAD

**Issue by usamashami11** [oneThousand1000/Portrait3D#8]
Hi, Great Work you guys have done!
I have some questions regarding the steps to run inference 

Currently, I have set up environment till this point, and now doing this part:
https://github.com/oneThousand1000/Portrait3D?tab=readme-ov-file#image-generation

1. However, I want to confirm if I have to clone this whole **Realistic_Vision_V5.1_noVAE** repo or download some specific weights like `Realistic_Vision_V5.1.ckpt` and `Realistic_Vision_V5.1.safetensors` as shown in the image?

![image](https://github.com/user-attachments/assets/c78448ea-54d8-47eb-98a7-b9c78c7b9204)

2. Secondly, to completely run the inference till the very last, where we acquire a 3D generated high quality portrait (like you showed in Gallery), do we have to run these as well? E.g.
- Image Processing:
https://github.com/oneThousand1000/Portrait3D?tab=readme-ov-file#image-processing
- Portrait Inversion
https://github.com/oneThousand1000/Portrait3D?tab=readme-ov-file#3d-portrait-inversion
- 3D Portrait Generation and Optimization
https://github.com/oneThousand1000/Portrait3D?tab=readme-ov-file#3d-portrait-generation-and-optimization

3. **Last**, is it possible to train/test your model for other objects, which are not human?

**Comment by oneThousand1000** [comment]
Hi,

1. On my machine, I downloaded all the folders and the Realistic_Vision_V5.1.ckpt file.
2. Yes, you need to run all three steps to generate a 3D avatar from a text prompt.
3. Our framework is theoretically suitable for general objects. However, the 3D GANs and diffusion models we've used are specifically trained for human generation. To adapt the framework for general objects, you'll need to replace these models with ones designed for those objects. Additionally, the 3D GANs you use would need to be retrained, as we've introduced a new representation. So, the process is somewhat complex.

**Comment by usamashami11** [comment]
Thanks for the prompt response. Really appreciate it!

> 1. On my machine, I downloaded all the folders and the Realistic_Vision_V5.1.ckpt file.

Thanks for telling, because the overall size of the complete Repo is too large. You may specify the specific file name (for hint)  for future users, although until now, the steps are quite easy to understand.

And okay, I'll try to run all the steps, and will ask you in case any issue is observed. Can I keep this issue open till I run the inference. You can close the issue after asking me, but I'll try to do the inference in this week, if everything goes smoothly.

> 3. Our framework is theoretically suitable for general objects. . . . we've introduced a new representation. So, the process is somewhat complex.

Got it. If all the training steps are given (which I believe you have, properly), I think I can give it a try after running the inference.

**Comment by oneThousand1000** [comment]
Feel free to keep this issue open. 
Have updated the readme.

**Comment by UsamaShami11** [comment]
Thank you!

**Comment by usamashami11** [comment]
Hi, I wanted to add some issues I observed and their solutions, that worked for me.

1. In this section (https://github.com/oneThousand1000/Portrait3D?tab=readme-ov-file#image-processing), while running `python -m pip install -e detectron2`, it gave me CUDA mismatch error (screenshot attached), due to the fact that my Machine had **NVIDIA (R) Cuda compiler driver version 10.1** (by default) and the environment `portrait3d_data` uses **CUDA Version 11.3** for compiling PyTorch as evident in this command and the screenshot, `pip install torch==1.10.2+cu113 torchvision==0.11.3+cu113 -f https://download.pytorch.org/whl/torch_stable.html`:

![Errorin Portrait 3D Screenshot 2024-08-20 101917](https://github.com/user-attachments/assets/2175f1ab-74eb-4084-b5aa-b5479be5ba7b)

**Solution:** Run `conda install nvidia/label/cuda-11.3.0::cuda-nvcc` to install CUDA Version 11.3

2. In this section (https://github.com/oneThousand1000/Portrait3D?tab=readme-ov-file#3d-portrait-inversion), while running this command (after setting your default paths),
```
python run_inversion_with_pose_optimization.py \
	--model_pkl=./models/model_512.pkl \
	--pose_prediction_kwargs_path=./models/model_512.json \
	--test_data_dir=../test_data \
	--inversion_name=final_inversion \
	--with_pose_optim
``` 
It gave me following error, due to the fact that my Machine had **NVIDIA (R) Cuda compiler driver version 10.1**  and the environment `text_to_3dportrait` uses **CUDA Version 11.6** for compiling PyTorch as evident in this command `pip install torch==1.12.1+cu116 torchvision==0.13.1+cu116 -f https://download.pytorch.org/whl/torch_stable.html`:

```
Setting up PyTorch plugin "bias_act_plugin"... Failed!
...
nvcc fatal   : Unsupported gpu architecture 'compute_86'
```
It can also be due to the version incompatibility of installed CUDA toolkit version with the `compute_86` architecture. I found that CUDA 11.0 and above support `compute_86`. (Here is a useful [GitHub issue](https://github.com/jkjung-avt/tensorrt_demos/issues/436) for the same.)

**Solution:** Run `conda install nvidia/label/cuda-11.6.0::cuda-nvcc` to install CUDA Version 11.6

--------------------------------------------------------------------------
**Pro-Tip:** You can check relevant NVIDIA (R) Cuda compiler driver version using `nvcc --version` command, and install the required missing nvcc version from this link: [NVIDIA cuda-nvcc packages](https://anaconda.org/nvidia/cuda-nvcc)
--------------------------------------------------------------------------


3. I'm facing one more issue while implementing the last step (https://github.com/oneThousand1000/Portrait3D?tab=readme-ov-file#results-gallery). I have downloaded examples from the Portrait3D_gallery, organized the directories and launched the visualizer app. However, when I input the paths to `Pickle` and `Pyramid Tri-Grid Ckpt`, it displays **Loading**, or sometimes error, **pickle or trigrid.pth not found**. (screenshot attached)

![Error in Portrait 3D Screenshot 2024-08-20 181212](https://github.com/user-attachments/assets/95e639f2-316e-44a1-a93e-5a538fc17213)

Can you tell me how much time it takes to visualize the image, after launching the app and setting up the paths? If it shows instantly, then what can be the possible issue? All other steps have been done, but only this step is left, so, any sort of assistance would be appreciated!

**Comment by oneThousand1000** [comment]
1&2. Yes, this project uses Cuda 11.6, which I mentioned in the requirements section of readme.
<img width="677" alt="image" src="https://github.com/user-attachments/assets/55d49cd0-c5a6-4924-8adb-d17ae636839b">
3. I guess there are some bugs in the visualizer. You should **input** paths to Pickle and Pyramid Tri-Grid Ckpt, and then **press enter (on your keyborad)** for actually loading those models. Hope that will be helpful. Will update the readme.

**Comment by oneThousand1000** [comment]
Anyway, thanks for your feedback, and I hope this issue will be helpful for others!

**Comment by usamashami11** [comment]
> 1&2. Yes, this project uses Cuda 11.6

I may have missed it, so apologies for that. Rest, I agree. Basically, in your virtual environments, `text_to_3dportrait` uses 11.6 while the `portrait3d_data` uses 11.3. So even if I have updated my default CUDA Version to 11.6, `portrait3d_data` may or may not cause issues (I think it won't as it was more of a `compute_86` issue, and it supports Version >= 11.0). 
Also, my purpose of highlighting was to ensure that one can get the same CUDA version for that particular environment, even if they don't have the desired CUDA version in their Machine (by default).

> 3. I guess there are some bugs in the visualizer. You should **input** paths to Pickle and Pyramid Tri-Grid Ckpt, and then **press enter (on your keyborad)** for actually loading those models. Hope that will be helpful. Will update the readme.

Alright. I have followed your method and it works! Thanks!

I may contact you again, in the future, if I need assistance in using this model for common objects, in context to which modules/models exactly need to be replaced, but I may do so over email, or by opening another issue.
For now, you can close this issue. **Thank you so much once again for your support and assistance!**
